lib/annotate/annotate_models.rb

```diff
@@ -362,7 +366,7 @@ def annotate_one_file(file_name, info_block, position, options = {})
       old_header = old_content.match(header_pattern).to_s
       new_header = info_block.match(header_pattern).to_s
 
-      column_pattern = /^#[\t ]+[\w\*\.`]+[\t ]+.+$/
+      column_pattern = /^#[\t ]+[\w\*\.\`\@\!\:]+[\t ]+.+$/
```

Comment:
I was originally going to break this regular expression up into something like:

```ruby
def column_pattern_for(format_type)
  case format_type
   when :yard then //
   when :rdoc then //
    # etc...
    end
end
```

By the time I go to the end, I saw that the original regular expression could be updated and cover all cases. 

I could see an advantage to breaking this up, but didn't want to go too far with refactoring and delay sharing this fix.

Reply:
@tmr08c if you're still interested in getting this patch in, I think there's value into breaking it up. The column pattern regex is hard to understand already.

Reply:
Thanks for the review and feedback @drwl! Working on updating this PR and adding some additional tests.